src/core/renderer/Renderer.cpp

```diff
@@ -87,26 +164,84 @@ namespace YimMenu
 		return ImGui_ImplWin32_WndProcHandler(hwnd, msg, wparam, lparam);
 	}
 
+	void Renderer::WaitForLastFrame()
+	{
+		std::this_thread::sleep_for(200ms);
+	}
```

Comment:
Is this called from multiple places that this is a separate function?

Reply:
An explanation on why this is needed would be very nice

Reply:
No, it is not. It's just to make things eaiser. And the reason was due to how DX12 resize buffers works. The overhead for doing it properly was not worth it. It would require 4-5 pointers and a lot of logic.

Reply:
https://github.com/ocornut/imgui/blob/master/examples/example_win32_directx12/main.cpp#L450C13-L454

https://github.com/ocornut/imgui/blob/master/examples/example_win32_directx12/main.cpp#L393-L407

if i'm not mistaken what they do can be entirely done within the ResizeBuffer hook

Reply:
Nope. That would always cause a crash, even when pulling from the game. I have no clue why. Not waiting would cause  deadlock in some cases. DX12 resizing is functionally different from DX11, but R* half-assily ported it to DX12

Reply:
This was the simplest solution with the least overhead (imo, that also worked the best)

Reply:
Fair enough i'm fine with that, but please add a small todo comment that say that this should be investigated further, a random sleep will certainly make some users crashes in one way or another when they resize

Reply:
I was already doing so as you said that lol.

Reply:
It's essentially the same thing but with no extra code overhead. Here we're assuming that 200ms is enough for a wait. And it most likely is overkill. It will certainly not cause a crash in my testing. And it's not even called that often since resize buffers only gets called one the user changes res or window type

Reply:
If anyone wants to work on this here's the ID3D12Fence pointer: ``4C 8B 05 ? ? ? ? 41 8B 88 + 3 rip`` as class and 0x10 is the ID3D12Fence. + 0x18 from the class is the FenceEvent HANDLE

Reply:
You never know what could be happening on the user machine, 200ms might not be enough. Anyway this is not a blocking thing so I'm fine if that get merged as is.

Reply:
> You never know what could be happening on the user machine, 200ms might not be enough. Anyway this is not a blocking thing so I'm fine if that get merged as is.

We can increase it to 1 second and it still wouldn't be noticeable. I doubt any machine would take longer then that as if it did it wouldn't even be capable of running RDR.

Reply:
How about meeting it halfway at 500. If this remains as the only topic, we can merge this.

Reply:
At a rate of 60 FPS, each frame would have a duration of approximately 16.67 milliseconds. So even 200ms is overkill IMO. Although i'm not exactly sure how Fence works but that's just my guess.